app/backend/db/crud.py

```diff
@@ -1,45 +1,54 @@
-import os
-from contextlib import contextmanager
-
-from sqlalchemy.orm import sessionmaker
-from sqlalchemy import create_engine
-
-from db import models
-
-
-USERNAME = os.environ.get('ONE_REPORT_USERNAME', 'one_report')
-PASSWORD = os.environ.get('ONE_REPORT_PASSWORD', 'one_report')
-HOST = os.environ.get('ONE_REPORT_HOST', 'localhost')
-DB = os.environ.get('ONE_REPORT_DB', 'one_report')
-PORT = os.environ.get('ONE_REPORT_PORT', '5432')
-
-DATABASE_URI = f'postgres+psycopg2://{USERNAME}:{PASSWORD}@{HOST}:{PORT}/{DB}'
-
-engine = create_engine(DATABASE_URI)
-Session = sessionmaker(bind=engine)
-
-
-@contextmanager
-def transaction():
-    s = Session()
-    try:
-        yield s
-
-    except:
-        s.rollback()
-        raise
-
-    finally:
-        s.close()
-
-
-def recreate_database():
-    with transaction() as s:
-        models.Base.metadata.drop_all(engine)
-        models.Base.metadata.create_all(engine)
-        mador = models.Mador(name='pie')
-        mador_settings = models.MadorSettings(
-            mador=mador, key='default_reminder_time', value='08:00', type='time')
-
-        s.add_all([mador, mador_settings])
-        s.commit()
+from sqlalchemy.orm import Session
+from datetime import date, time
+from typing import List
+
+from .models import User, DateData
+from utils.datetime_utils import daterange
+from db import schemas
+
+# User:
+
+def get_user(db: Session, user_id: int) -> User:
+    return db.query(User).filter(User.id == user_id).one()
+
+def get_subjects(db: Session, commander_id: int) -> List[User]:
+    commander = get_user(db, commander_id)
+    return commander.soldiers
+
+def get_reminder(db: Session, user_id:int) -> time:
+    user = get_user(db, user_id)
+    return user.reminder_time
+
+def was_reminded(db: Session, user_id: int) -> bool:
+    user = get_user(db, user_id)
+    return user.last_reminded_date == date.today()
+
+# Date Data:
+
+def get_dates_data(db: Session, user_id: int, 
+                   start_date: date, end_date: date = None) -> List[DateData]:
+    return db.query(DateData).filter(DateData.user_id == user_id, 
+                                     start_date <= DateData.date <= end_date).all()
```

Comment:
you set end_date as default value None.
and then u compared 
date <= None 
fix it

Reply:
Done